Update username in EIP-1319

njgheorghita opened the pull request:
Added the specification for ethPM uris. They've been an effective solution for handling packages so far - and are already implemented in the `ethpm-cli` and `Brownie`, and are also outlined in the [documentation](https://docs.ethpm.com/uris#registry-uris).

eip-automerger commented:
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - EIP 1319 requires approval from one of (christophergewecke@gmail.com, nick.dandrea@consensys.net, piper@ethereum.org, nickg@ethereum.org)

njgheorghita commented:
ping @gnidan @iamdefinitelyahuman @pipermerriam for input/confirmation. Specifically if all looks well, could @gnidan approve this pr - since my email listed in the pr is not the one associated with my github, so I can't get this merged.

iamdefinitelyahuman commented:
I like it!  Looks very clean.  Thanks for pinging me on this.

pipermerriam commented on EIPS/eip-1319.md:
```diff
@@ -166,6 +166,62 @@ release versions while allowing them to use any versioning schema they choose.
+- Optional
+- String of the target package version
+- If the package version contains any [url unsafe characters](https://www.werockyourweb.com/url-escape-characters/), they **must** be safely escaped
+- Since semver is not strictly enforced by the ethpm spec, if the `package_version` is omitted from a uri, it is up to the tooling / framework to look up all available versions and ask the user which version they want to utilize.
```
This would probably be better to use the `**should**` language.  Something like:

> if the `package_version` is omitted from a uri, tooling **should** avoid guessing in the face of any ambiguity and present the user with a choice from the available versions.

pipermerriam commented on EIPS/eip-1319.md:
```diff
@@ -166,6 +166,62 @@ release versions while allowing them to use any versioning schema they choose.
+
+#### `registry_address`
+- Required
+- This **should** be an ENS name whenever possible, or a `0x`-prefixed, checksummed address. 
```
This reads as if an `ENS` name is preferable to the other formats.  Maybe reword as 

> This **should** be either an ENS name or a `0x`-prefixed, checksummed address.   ENS names are more suitable for cases where mutability of the underlying asset is acceptable and there is implicit trust in the owner of the name.  `0x` prefixed addresses are more preferable in higher security cases to avoid needing to trust the controller of the name.

axic commented:
@njgheorghita is this still outstanding for merging?

njgheorghita commented:
@axic Thanks for the reminder. It has all the necessary approvals afaik, it's just that the email address associated with my github account is not the one listed under my name as an author on this ERC. Would you be able to push the merge through?

axic commented:
@njgheorghita do you want to change the author line to github username or another email so you can merge these draft changes?

axic commented:
Can you also merge in master (or rebase)? We should check what the validator says now as this was creating quite a while ago.

axic commented on EIPS/eip-1319.md:
```diff
@@ -166,6 +166,63 @@ release versions while allowing them to use any versioning schema they choose.
 
 Registries may offer more complex `read` APIs that manage requests for packages within a semver range or at `latest` etc. This EIP is agnostic about how tooling or registries might implement these. It recommends that registries implement [EIP 165](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-165.md) and avail themselves of resources to publish more complex interfaces such as [EIP 926](ht ...
 
+## EthPM URIs
```
I'm not fully sure the best way is to include both the URI format and everything else in a single EIP. Would suggest for consideration to make this a separate EIP depending on this one.

njgheorghita commented:
@axic I've gone ahead and moved the Ethpm URI spec to a new [ERC](https://github.com/ethereum/EIPs/pull/2942). I updated my username in this ERC just in case there are future edits that need to be made.

MicahZoltu commented:
Should probably make this EIP require `2942`, since it references the path specification in here.

MicahZoltu commented:
Though, it currently only references it casually in an example, without making any other mention anywhere in the specification of it.  Maybe we can just remove that one example and merge this (which ends up being just an author update)?

njgheorghita commented:
@MicahZoltu Agreed, made the change, thanks!
